tests/bdd/volgroup_vgpattern_bdd.feature

```diff
@@ -0,0 +1,131 @@
+
+Feature: Validate vgpattern and volgroup based scheduling.
+  Background:
+      Given Single lvmnode cluster is configured
+      And lvm vg "lvmvg" is created on a node
+
+      Scenario: test sc with volgroup present on the node
+          Given a sc is created with volgroup as "lvmvg"
+          And pvc is created referencing the same sc
+          Then lvmvolume cr should get created
+          And cr state should be Ready
+
+      Scenario: test sc with volgroup not present on the node
+          Given a sc is created with volgroup as "lvmvgn"
+          And pvc is created referencing the same sc
+          Then pvc should be in Pending
+          And lvmvolume cr should not get created
+
+  
+  Background:
+      Given Single lvmnode cluster is configured
+      And lvm vg "lvmvg" and "lvmvg1" is created on a node
+
+      Scenario: test sc with vgpattern matching atleast one vg
+          Given sc is created with vgpattern "lvmvg1.*"
+          When a pvc is created by referencing the sc
+          Then pvc should be Bound
+          And lvmvolume should be in "lvmvg1" vg
+
+  
+      Scenario: test sc with vgpattern matching no vg
+          Given sc is created with vgpattern "lvmvgn.*"
+          When a pvc is created by referencing the sc
+          Then pvc should not be Bound
+          And lvmvolume should not be created
+
+  
+      Scenario: test pvc with higher size then vg free space
+          Given "lvmvg" has lesser free space then "lvmvg1"
+          And sc is created with volgroup as "lvmvg"
+          When a pvc is created with size higher then "lvmvg" and lower then "lvmvg1" free space
+          Then pvc should not be boun
+          And lvmvolume cr should not be created
+
+  
+  Background:
+      Given a 3 lvmnode cluster is configured
+      And each node <node> has <vg> created with same size 
+
+      Scenario: test sc with volgroup present only on one node
+          Given a sc is created with volgroup as "lvmvg1" with no topology
+          When pvc is created referencing the same sc
+          Then pvc should be in Pending state
+          And lvmvolume cr should not be created
```

Comment:
Well it might not always be pending? What if it selects the node that has the vg?

Reply:
We need a variation for this with vgpattern?

Reply:
I thought so initially... However, when i was testing manually. It always selects a node which doesnt have matching vg. In between retires of i created vg on node when it failed in the previous attemt. It doesnt pick that node.

I added this case with expectation that it will fail sometimes. But doesnt seem like happening. If it does also i think let it Fail. That would be an inconsistent behaviour.

Reply:
Are we discouraging users via documentation to not use vgpattern and volgroup both in the the storage class. We should do that I think even though we somehow handle both being present in sc.

Reply:
Currently we have not documented. But we should. There is an enhancement planned to return error during /`CreateVolume` if both are supplied and they dont compliment each other.

Incase both are supplied then volgroup is considered for scheduling.